### ISSUE TITLE
Lint messages reported from porter should not mention mixins

### ISSUE THREAD

**Issue by carolynvs** [getporter/porter#2526]
When porter generates a lint message directly, instead of the message coming from a mixin, we should change the message so that it doesn't print out extra "empty" text about mixins that isn't relevant.

Once #2434 is merged, you can do the following to reproduce:

1. porter create
2. edit porter.yaml and define a parameter named "porter_test"
3. porter lint

expected output
```
Running linters for each mixin used in the manifest...
error(porter-100) - Reserved name warning
porter_test has a reserved prefix. Parameters cannot start with porter- or porter_
See https://getporter.org/reference/linter/#porter-100 for more information
```

actual output
```
Running linters for each mixin used in the manifest...
error(porter-100) - Reserved name warning
: 0th step in the  mixin ()
porter_test has a reserved prefix. Parameters cannot start with porter- or porter_
See https://getporter.org/reference/linter/#porter-100 for more information
```

You can tell that a lint result was generated by porter because Location.Mixin is an empty string. The relevant code to change is 

https://github.com/getporter/porter/blob/6df27907b5a40482ceb8c036fede2fa56a518c16/pkg/linter/linter.go#L113-L116

Please update the unit tests in linter_test.go to validate that lint results generated by porter do not print out that extra text ": 0th step in the  mixin ()"

See our [Contributing Tutorial](https://getporter.org/contribute/tutorial) and [New Contributor Guide](https://github.com/getporter/porter/blob/release/v1/CONTRIBUTING.md) for help getting started contributing to Porter.

**Comment by 0xquark** [comment]
Hi @carolynvs, I'd like to work on this!

**Comment by carolynvs** [comment]
@0xquark That would be great! You won't be able to reproduce the behavior listed in the reproduction steps yet, because we haven't merged #2434 as mentioned in the issue above. But you can start on it now, and it should be merged soon. 👍

**Comment by 0xquark** [comment]
Thanks! I'll start working on this.
